pkg/xds/auth/callbacks.go

```diff
@@ -87,12 +88,16 @@ func (a *authCallbacks) OnStreamRequest(streamID core_xds.StreamID, req util_xds
 		return err
 	}
 
-	credential, err := ExtractCredential(s.ctx)
-	if err != nil {
-		return errors.Wrap(err, "could not extract credential from DiscoveryRequest")
-	}
-	if err := a.authenticator.Authenticate(user.Ctx(s.ctx, user.ControlPlane), s.resource, credential); err != nil {
-		return errors.Wrap(err, "authentication failed")
+	// If the gRPC stream is already authenticated, we don't need to authenticate it again.
```

Comment:
We can't put it in `OnStreamOpen` because we don't have the `resource` object in there.